src/pages/PortForwarding.vue

```diff
@@ -8,28 +8,53 @@
     :include-kubernetes-services="settings.portForwarding.includeKubernetesServices"
     :k8s-state="state"
     :kubernetes-is-disabled="!settings.kubernetes.enabled"
+    :service-being-edited="serviceBeingEdited"
+    :error-message="errorMessage"
+    @updatePort="handleUpdatePort"
     @toggledServiceFilter="onIncludeK8sServicesChanged"
+    @editPortForward="handleEditPortForward"
+    @cancelPortForward="handleCancelPortForward"
+    @cancelEditPortForward="handleCancelEditPortForward"
+    @updatePortForward="handleUpdatePortForward"
   />
 </template>
 
-<script>
+<script lang="ts">
 import { ipcRenderer } from 'electron';
+import Vue from 'vue';
 import PortForwarding from '@/components/PortForwarding.vue';
-import { defaultSettings } from '@/config/settings';
+import { defaultSettings, Settings } from '@/config/settings';
+import * as K8s from '@/k8s-engine/k8s';
 
-/** @typedef { import("../config/settings").Settings } Settings */
-
-export default {
+export default Vue.extend({
   components: { PortForwarding },
   data() {
     return {
-      state:         ipcRenderer.sendSync('k8s-state'),
-      /** @type Settings */
-      settings:      defaultSettings,
-      services: []
+      state:              ipcRenderer.sendSync('k8s-state'),
+      settings:           defaultSettings as Settings,
+      services:           [] as K8s.ServiceEntry[],
+      errorMessage:       null as string | null,
+      serviceBeingEdited: null as K8s.ServiceEntry | null,
     };
   },
 
+  watch: {
+    services(newServices: K8s.ServiceEntry[]): void {
+      // Typescript was complaining about simply using this.serviceBeingEdited,
```

Comment:
![image](https://user-images.githubusercontent.com/19395776/180077816-04489102-31c9-4bc6-801a-eaf490cb47ec.png)

If you can figure out a better way around it, I'm all ears.

Reply:
`compareServices` doesn't accept a nullable type and `serviceBeingEdited` can be `K8s.ServiceEntry | null`

It looks like something about `Object.assign({}, this.serviceBeingEdited);` is being interpreted as `K8s.ServiceEntry`, but it would actually end up being an empty object when `serviceBeingEdited` is null..   

This means that `if (localServiceBeingEdited)` will always evaluate to true and I don't think that's intentional. I would probably look at [using type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates), but I don't think what you have here will break at all. We just seem to be losing some assurance that typescript provides..

Reply:
I honestly think this is a bug with my setup. I solved it by doing `this.serviceBeingEdited as K8s.ServiceEntry`. Either way, it should be fine since this is all inside an if statement that checks that `this.serviceBeingEdited` is truthy.

Reply:
This thread was more of a curiosity to me than anything else. Much agreed that the implementation is fine 🙂